app/assets/javascripts/components/Register.jsx

```diff
@@ -2,99 +2,77 @@ class Register extends React.Component {
 
   constructor(props) {
     super();
-    this.state = {success: null};
+    this.state = {
+      success: '',
+      email: '',
+      password: '',
+      first_name: '',
+      last_name: '',
+      avatar: '',
+    };
   }
 
-  FieldGroup({ id, label, help, ...props }) {
-    const { Checkbox, Radio, FormGroup, ControlLabel, FormControl, Button, HelpBlock } = ReactBootstrap;
-    return (
-      <FormGroup controlId={id}>
-        <ControlLabel>{label}</ControlLabel>
-        <FormControl {...props} />
-        {help && <HelpBlock>{help}</HelpBlock>}
-      </FormGroup>
-    );
+  handleChange = (evt) => {
+    this.setState({ [evt.target.name]: evt.target.value });
+  }
+
+  setFile = (e) => {
+    const files = e.target.files;
+    if (!files || !files[0]) {
+      return
+    }
+
+    const reader = new FileReader();
+    reader.onload = (file) => {
+      this.setState({ avatar: file.target.result, });
+    }
+
+    reader.readAsDataURL(files[0]);
   }
 
   createUser = (evt) => {
-    evt.preventDefault()
-    let upload = "";
-    if (this.file && this.file.files.length > 0) {
-      upload = this.file.files[0];
+    evt.preventDefault();
+    const user = {
+      email: this.state.email,
+      password: this.state.password,
+      first_name: this.state.first_name,
+      last_name: this.state.last_name,
+      avatar: this.state.avatar
     }
-    const payload = {
-      email: this.email,
-      password: this.password,
-      first_name: this.first,
-      last_name: this.last,
-      avatar: upload
-    };
-    Requester.post('/api/users', payload).then((data) => {
-      this.setState({success: 1});
+    const params = {
+      user: user
+    }
+    Requester.post('/api/users', params).then((data) => {
```

Comment:
i think @lbkchen mentioned in @SaladRaider's pr, but we should put all routes like  `'/api/users'` in a separate `routes.jsx` file